### PR TITLE
Redesign the team gallery using sphinx-design's card directive

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Use [sphinx-design's card directive](https://sphinx-design.readthedocs.io/en/latest/cards.html) instead of the unreadable raw HTML codes added in #1308.

The `card` directive can also be used in https://github.com/GenericMappingTools/pygmt/issues/1936.

### CODE REVIEW

File: doc/team.md [hunk 1]
```diff
@@ -22,78 +22,64 @@ responsibilities. Distinguished Contributors who are also active maintainers of
...
+[@leouieda](https://github.com/leouieda)
+::::
+
+::::{grid-item-card} Liam Toney
+:img-top: https://avatars.githubusercontent.com/u/38269494?v=4
+
+[@liamtoney](https://github.com/liamtoney)
+::::
+
+::::{grid-item-card} Meghan Jones
```
Comment (seisman) [inline doc/team.md:52]:
```suggestion
::::{grid-item-card} Max Jones
```

File: doc/team.md [hunk 2]
```diff
@@ -22,78 +22,64 @@ responsibilities. Distinguished Contributors who are also active maintainers of
...
+
+Founder
+[@PaulWessel](https://github.com/PaulWessel)
+::::
+
+::::{grid-item-card} Wei Ji Leong
+:img-top: https://avatars.githubusercontent.com/u/23487320?v=4
+
+Maintainer
+[@weiji14](https://github.com/weiji14)
```
Comment (meghanrjones) [inline doc/team.md:77]:
```suggestion
[@weiji14](https://github.com/weiji14)
+++
Maintainer
```
| Before | After |
|-------|------|
| 
image | image |

File: doc/conf.py [hunk 3]
```diff
@@ -29,13 +29,15 @@
...
     "sphinx_copybutton",
+    "sphinx_design",
 ]
 
 # Autosummary pages will be generated by sphinx-autogen instead of sphinx-build
 autosummary_generate = []
 
 # Auto-generate header anchors with MyST parser
 myst_heading_anchors = 4
+myst_enable_extensions = ["colon_fence"]
```
Comment (weiji14) [inline doc/conf.py:40]: Document a bit what this does (enable `::::` code fences). https://myst-parser.readthedocs.io/en/v0.17.2/syntax/optional.html#code-fences-using-colons

```suggestion
# Allow code fences using colons
myst_enable_extensions = ["colon_fence"]
```

**Review by meghanrjones — COMMENTED** [review]
Thanks @seisman, so much simpler! One suggestion is to use the footer option so that the founder/maintainer labels are separated from the GitHub username (example provided as an in-line comment)

File: doc/team.md (hunk 2 above)
Comment (seisman) [reply doc/team.md]: Good suggestion! I've used the footer option for labels and also use [sphinx-design's badge directive](https://sphinx-design.readthedocs.io/en/latest/badges_buttons.html) to highlight different roles.

image

File: doc/team.md [hunk 4]
```diff
@@ -22,78 +22,64 @@ responsibilities. Distinguished Contributors who are also active maintainers of
...
-        <img class="card-img-top team-img" src="https://avatars.githubusercontent.com/u/29518865?v=4" alt="Card image">
-        <div class="team-card-body">
-            <h6 class="team-card-text">Will Schlitzer</h6>
-            <h7 class="team-card-text">Maintainer</h7>
-            <p class="team-card-text"><a href="https://github.com/willschlitzer" >@willschlitzer</a></h7>
-        </div>
-    </div>
-    </div>
-```
+:::::{grid} 2 3 3 4
```
[the remark this answers is not in the document]
Comment (seisman) [reply doc/team.md:25]: > :::::{grid} 2 3 3 4

These four numbers are the number of columns for extra-small, small, medium and large screens.

**Comment by seisman** [comment]
Here is the latest screenshot:
image

**Review by weiji14 — APPROVED** [review]
Very nice, the badges are a nice touch!

**Review by meghanrjones — APPROVED** [review]
Very nice!

**Pull request merged**